courses/utils.py

```diff
@@ -344,3 +345,30 @@ def get_catalog_course_filter(relative_filter=""):
         Q(**courseware_live_filter)
         & Q(Q(**courserun_start_date_filter) | Q(**courserun_enrollment_end_filter))
     )
+
+
+def get_catalog_languages():
+    """
+    Returns the languages that are associated with courses or programs visible in the catalog
+    """
+
+    course_languages = (
+        CourseLanguage.objects.filter(
```

Comment:
I am a bit confused:
When you asked:

> I explicitly added is_active field to the CourseLanguage model in the feature PR https://github.com/mitodl/mitxpro/pull/3384 solely to hide the languages from the catalog when needed. Now we are changing the implementation to hide languages automatically based on associated courses. Do you still feel the need for this?

Bon Replied with:
> I think we should keep this in case we need to turn off the language filter in it's entirety

I think Bon misunderstood this with the Posthog flag. Plus, if he meant to keep the per-language is_active field, then we are not checking that here.

Reply:
I missed that. Looks good.